### DISCUSSION TITLE
Error memoria con malloc

### DISCUSSION THREAD

**Discussion by rocimarquez** [IIC2133-PUC/2024-2#70] [Tarea 0] [answered]
Holaaaa, estaba probando con los test y en el Part-2-medium-2 me arroja un error de realloc al agregar puffles cuando se compran. Eso no me había pasado en los test anteriores y no se si estoy creando los puffles de manera incorrecta o utilizando puffles que se crearon mal. Busqué en internet y dice que se debe a errores de fuga de memoria, pero al ser un output tan largo es demasiado dificil identificarlo. Aqui dejo lo que me arrojó valgrind. Mi repo esta actualizado. --109565-- VALGRIND INTERNAL ERROR: Valgrind received a signal 11 (SIGSEGV) - exiting
--109565-- si_code=1;  Faulting address: 0x504B8225C;  sp: 0x1002ca9e30

valgrind: the 'impossible' happened:
   Killed by fatal signal

host stacktrace:
==109565==    at 0x5804CA1A: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==109565==    by 0x58005681: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==109565==    by 0x5809B03A: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==109565==    by 0x580E3F40: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)

sched status:
  running_tid=1

Thread 1: status = VgTs_Runnable (lwpid 109565)
==109565==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==109565==    by 0x109A07: agregar_puffle (main.c:161)
==109565==    by 0x109B50: comprar_puffle (main.c:184)
==109565==    by 0x10A6E6: main (main.c:412)
client stack range: [0x1FFEFFD000 0x1FFF000FFF] client SP: 0x1FFEFFFB40
valgrind stack range: [0x1002BAA000 0x1002CA9FFF] top usage: 18744 of 1048576

**Comment by sebacarrascop** [comment, marked as answer]
hola!, si pudiste pasar Part-2-medium-1 sin problema, pero no medium 2, lo más probable que tu codigo se cae en algún caso borde de las funciones que se testean con medium. Me enfocaría principalmente en GIVE_PUFFLE, ya que TRADE_PUFFLE y STEAL_PUFFLE se testean desde Part-2-easy. Saludos.
